polus-basic-flatfield-correction-plugin/src/main.py

```diff
@@ -56,50 +56,39 @@ def main():
     logger.info('get_photobleach = {}'.format(get_photobleach))
     logger.info('inp_regex = {}'.format(inp_regex))
     logger.info('output_dir = {}'.format(output_dir))
-    test = FilePattern(fpath, inp_regex)
-    files = [i for i in test.get_matching()]
     # Set up lists for tracking processes
     processes = []
     process_timer = []
     pnum = 0
-    rs = list (set([b for key in files for a,b in key.items() if a =='r']))
-    rs.sort()
-    for r in rs:
-        ts = list (set([b for key in files for a,b in key.items() if a =='t']))
-        ts.sort()
-        for t in ts:
-            cs = list (set([b for key in files for a,b in key.items() if a =='c']))
-            cs.sort()
-            for c in cs:
-                # The optimization process seems to use up to ~6 cores, so limit # of processes accordingly
-                if len(processes) >= multiprocessing.cpu_count() - 1:
-                    free_process = -1
-                    while free_process < 0:
-                        for process in range(len(processes)):
-                            if processes[process].poll() is not None:
-                                free_process = process
-                                break
-                        # Wait between checks to free up some processing power
-                        time.sleep(3)
-                    pnum += 1
-                    logger.info("Finished process {} of {} in {}s!".format(pnum, len(rs) * len(ts) * len(cs),
+    file,uval=parse_directory(fpath,inp_regex)
+    if len(processes) >= multiprocessing.cpu_count() - 1:
+        free_process = -1
+        while free_process < 0:
+            for process in range(len(processes)):
+                if processes[process].poll() is not None:
+                    free_process = process
+                    break
+            # Wait between checks to free up some processing power
+            time.sleep(3)
+        pnum += 1
+        logger.info("Finished process {} of {} in {}s!".format(pnum, len(uval['r']) * len(uval['t']) * len(uval['c']),
```

Comment:
I don't see any kind of loop here to loop over sets of data. Will this code ever generate more than one process?